lib/fileHandler/index.js

```diff
@@ -335,19 +318,40 @@ module.exports = function (options) {
 		};
 	}
 
+	//Check if the URL and response is a pre-signed URL from AWS
 	function checkExpiryConditions (url, headers, rawBody) {
 		return (
-			_.includes(url, '.amazonaws.com/') &&
+			( _.includes(url, 'Expires=') || _.includes(url, 'X-Amz-Date=') )&&
 			_.includes(headers['content-type'], 'application/xml') &&
 			_.includes(headers.server, 'AmazonS3') &&
 			_.includes(rawBody.toString(), 'Request has expired')
 		);
 	}
 
-	function expiryRejectObject () {
+	function expiryRejectObject (url, headers) {
+		const payload = {};
+		const targetUrl = new URL(url);
+
+		if (targetUrl.searchParams.has('Expires')) {
+			const expiryTime = targetUrl.searchParams.get('Expires');
+			if (expiryTime) {
+				//snake_case as this is exposed to the user
+				payload.datetime_expired = moment(targetUrl.searchParams.get('Expires'), 'X').format();
+			}
+		}
+
+		if (targetUrl.searchParams.has('X-Amz-Date')) {
+			const expiryDate = targetUrl.searchParams.get('X-Amz-Date');
+			if (expiryDate) {
+				//snake_case as this is exposed to the user
+				payload.datetime_expired = moment(targetUrl.searchParams.get('X-Amz-Date'), 'YYYYMMDD[T]HHmmss[Z]').format();
```

Comment:
Looking at this page - https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html it looks you have to add `X-Amz-Expires` to `X-Amz-Date`to get the expiration time